src/accounts/api/views.py

```
@@ -181,3 +184,45 @@ def post(self, request):
         public_data = PublicAccountDataSerializer(account).data
 
         return Response(public_data, status=status.HTTP_200_OK)
+
+
+class ResetPasswordView(GenericAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = ResetPasswordSerializer
+
+    def post(self, request, reset_token):
+        serializer = self.serializer_class(data=request.data)
+        try:
+            if serializer.is_valid(raise_exception=True):
+                reset_request = PasswordResetRequestModel.objects.get(token=reset_token)
+                if not reset_request.is_token_valid():
+                    return Response({"error": "Invalid link or expired."}, status=status.HTTP_400_BAD_REQUEST)
```

`is_token_valid` only fails when db record is expired, so should probably say `Expired token`

---

I don't have an issue with it saying invalid or expired. Now, I think it should instead raise an exception so it is handled with the rest of errors.